src/soc/ibm/power9/chip.c

```diff
@@ -1,6 +1,75 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
 #include <device/device.h>
+#include <fit.h>
+#include <cpu/power/istep_13.h>
+#include "istep_13_scom.h"
+
+static int dt_platform_fixup(struct device_tree_fixup *fixup,
+			      struct device_tree *tree)
+{
+	struct device_tree_node *node;
+
+	/* Memory devices are always direct children of root */
+	list_for_each(node, tree->root->children, list_node) {
+		const char *devtype = dt_find_string_prop(node, "device_type");
+		if (devtype && !strcmp(devtype, "memory")) {
+			dt_add_u32_prop(node, "ibm,chip-id", 0 /* FIXME for second CPU */);
+		}
+	}
+
+	return 0;
+}
+
+#define SIZE_MASK	PPC_BITMASK(13,23)
+#define SIZE_SHIFT	(63 - 23)
+#define BASE_MASK	PPC_BITMASK(24,47)
+#define BASE_SHIFT	(63 - 47)
+
+/* Value in registers is in 4GB units, ram_resource() expects kilobytes. */
+#define TO_KB(x)	((x) << 22)
+
+static inline unsigned long base_k(uint64_t reg)
+{
+	return TO_KB((reg & BASE_MASK) >> BASE_SHIFT);
+}
+
+static inline unsigned long size_k(uint64_t reg)
+{
+	return TO_KB(((reg & SIZE_MASK) >> SIZE_SHIFT) + 1);
+}
+
+static void enable_soc_dev(struct device *dev)
+{
+	int mcs_i, idx = 0;
+
+	for (mcs_i = 0; mcs_i < MCS_PER_PROC; mcs_i++) {
+		uint64_t reg;
+		chiplet_id_t nest = mcs_to_nest[mcs_ids[mcs_i]];
+
+		/* These registers are undocumented, see istep 14.5. */
+		reg = read_scom_for_chiplet(nest, 0x0501080A);
```

Comment:
Acording to hosboot code, this register is a `Primary MemCfg Reg`, `MCFGP0` or `P9 MCS target MEMCFG REG`. Please consider naming it something with like that.

Reply:
In previous steps i kept undocumented registers addresses as a value, so I think it would be better to keep it here too. I added a comment with probable name of the register.